src/lib/dataApi.ts

```diff
@@ -94,38 +94,44 @@ export class DataApi {
     );
   }
 
-  async createNote(record: DataRecord, templatePath: string): Promise<void> {
-    let content = "";
+  async createNote(record: DataRecord, templatePath: string, useTemplater: boolean): Promise<void> {
+    const templater = getTemplater();
+    let file;
+    let content;
 
     if (templatePath) {
-      const file = this.fileSystem.getFile(templatePath);
-      if (file) {
-        content = await file.read();
-        content = interpolateTemplate(content, {
-          title: () => getNameFromPath(record.id),
-          date: (format) => moment().format(format || "YYYY-MM-DD"),
-          time: (format) => moment().format(format || "HH:mm"),
-        });
-        if (record.values["tags"]) {
-          const templateTags = F.pipe(
-            content,
-            decodeFrontMatter,
-            E.map((frontmatter) => frontmatter["tags"]),
-            E.fold(
-              () => [],
-              (right) => right ?? [] // handle `null`
-            )
-          );
-          //@ts-ignore explict input in `createDataRecord()`
-          const tagSet: Set<string> = new Set(
-            templateTags.concat(record.values["tags"])
-          );
-          record.values["tags"] = [...tagSet];
-        }
-      }
+      content = await this.fileSystem.getFile(templatePath)?.read();
     }
 
-    const file = await this.fileSystem.create(record.id, content);
+    if (!content) {
+      file = await this.fileSystem.create(record.id, "");
+    } else if (useTemplater && templater) {
+      const newTemplaterFile = await templater.templater.create_new_note_from_template(content, record.path, record.name, false);
+      file = this.fileSystem.getFile(newTemplaterFile.path)!;
+    } else {
+      content = interpolateTemplate(content, {
+        title: () => getNameFromPath(record.id),
+        date: (format) => moment().format(format || "YYYY-MM-DD"),
+        time: (format) => moment().format(format || "HH:mm"),
+      });
+      if (record.values["tags"]) {
+        const templateTags = F.pipe(
+          content,
+          decodeFrontMatter,
+          E.map((frontmatter) => frontmatter["tags"]),
+          E.fold(
+            () => [],
+            (right) => right ?? [] // handle `null`
+          )
+        );
+        //@ts-ignore explict input in `createDataRecord()`
+        const tagSet: Set<string> = new Set(
+          templateTags.concat(record.values["tags"])
+        );
+        record.values["tags"] = [...tagSet];
```

Comment:
Quick explaination on L117-L131.

There's a certain case that, when user create a new note under a tag-sourced project, the tag is expected to added to the frontmatter automatically. It's neccessary to uniquify the `tags` field if there's already one listed in the template. See #646 

So in the `useTemplater` branch we hope it can act the same.